src/test/java/de/tum/in/www1/artemis/service/connectors/TextSimilarityClusteringServiceTest.java

```diff
@@ -42,18 +64,25 @@ public void clusterTextBlocks() throws NetworkingError {
         ReflectionTestUtils.setField(textEmbeddingService, "API_ENDPOINT", EMBEDDING_ENDPOINT);
         final List<TextEmbedding> embeddings = textEmbeddingService.embedTextBlocks(blocks, exercise);
 
-        final Map<Integer, TextCluster> clusterDictionary = service.clusterTextBlocks(embeddings);
+        final TextSimilarityClusteringService.Response response = service.clusterTextBlocks(embeddings, 3);
+        final Map<Integer, TextCluster> clusterDictionary = response.clusters;
 
-        assertThat(clusterDictionary.keySet(), hasSize(1));
+        assertThat(clusterDictionary.keySet(), hasSize(5));
         assertThat(clusterDictionary.keySet(), hasItem(-1));
-        final TextCluster cluster = clusterDictionary.get(-1);
-        final List<TextBlock> blocks1 = cluster.getBlocks();
 
-        assertThat(blocks1.toArray(), is(equalTo(blocks.toArray())));
-
-        final double[][] distanceMatrix = cluster.getDistanceMatrix();
-        assertThat(distanceMatrix[0][1], is(equalTo(distanceMatrix[1][0])));
-        assertThat(distanceMatrix[0][1], is(both(greaterThan(0.5)).and(lessThan(0.7))));
+        List<List<Double>> matrix = response.distanceMatrix;
+        assertThat(matrix, hasSize(blocks.size()));
+
+        List<TreeNode> clusterTree = response.clusterTree;
+        List<TreeNode> blocksInTree = clusterTree.stream().filter(treeNode -> treeNode.isBlockNode()).collect(Collectors.toList());
+        // Assert that number of blockNodes in the tree equals number of blocks
+        assertThat(blocksInTree, hasSize(blocks.size()));
+        List<Long> groupByChild = clusterTree.stream().map(treeNode -> treeNode.getChild()).distinct().collect(Collectors.toList());
+        // Assert that child is a unique property of a TreeNode
+        assertThat(groupByChild, hasSize(clusterTree.size()));
+        List<Long> groupByParent = clusterTree.stream().map(treeNode -> treeNode.getParent()).distinct().collect(Collectors.toList());
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'Collectors.toList' due to existing static import 'java.util.stream.Collectors.toList'](https://app.codacy.com/gh/ls1intum/Artemis/pullRequest?prid=5968857)